woocommerce/google-listings-and-ads
# Formatting with Prettier

### Changes proposed in this Pull Request:

This PR does the following: 

- Adds a new `.prettierignore` file. See the comments in the file for the reasons to ignore them. For more info, see https://prettier.io/docs/en/install.html.
- Adds additional configuration in `.editorconfig` file. It is pretty much similar to https://github.com/woocommerce/woocommerce-admin/blob/1e92ba6ef397563aa6abfda1ad30daadd640b1ac/.editorconfig.
- Rename .stylelintrc to .stylelintrc.json. This is to make prettier able to recognize it as json file and format it accordingly.
- Contains the output of running `npx prettier --write .` command.

Prior to this PR, I was using [VS Code ESLint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) as the default formatter in VS Code. I have since changed to using [Prettier formatter](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode). It works with more file types, and - like ESLint extension - it will auto format the files when I hit save.

### Detailed test instructions:

1. Install Prettier formatter in your VS Code / editor. Enable auto format on save.
2. Checkout this branch.
3. Open some files that are not ignored by Prettier (e.g. JS, JSON, MD files) and make some modifications. Upon save, it should auto format the file.
4. Open files that are ignored by Prettier (e.g. CSS, SCSS, yml files) and make some modifications. Upon save, it should **not** auto format the file.
    - Note: CSS and SCSS files should be auto formatted by stylelint.

### Changelog entry

> Fix - Code formatting with Prettier.
